src/wallet/rpc/wallet.cpp

```diff
@@ -809,6 +809,61 @@ static RPCHelpMan migratewallet()
     };
 }
 
+RPCHelpMan getxpub()
+{
+    return RPCHelpMan{"getxpub",
+                "Returns the xpub most recently used to generate descriptors for this descriptor wallet. "
+                "Not entirely useful right now as it returns the xpub of the root, and there are "
+                "hardened derivation steps involved in normal key derivation.\n",
+                {
+                    {"private", RPCArg::Type::BOOL, RPCArg::Default{false}, "Whether to include the xprv"},
+                },
+                RPCResult{
+                    RPCResult::Type::OBJ, "", "",
+                    {{
+                        {RPCResult::Type::STR, "xpub", "The xpub"},
+                        {RPCResult::Type::STR, "xprv", /*optional=*/true, "The xprv if private is true"},
+                    }},
+                },
+                RPCExamples{
+                    HelpExampleCli("getxpub", "")
+            + HelpExampleRpc("getxpub", "")
+                },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+        {
+            const std::shared_ptr<CWallet> pwallet = GetWalletForJSONRPCRequest(request);
+            if (!pwallet) return NullUniValue;
+
+            if (!pwallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS)) {
+                throw JSONRPCError(RPC_WALLET_ERROR, "getxpub is not available for non-descriptor wallets");
+            }
+
+            LOCK(pwallet->cs_wallet);
+
+            UniValue obj(UniValue::VOBJ);
+            std::optional<CExtPubKey> extpub = pwallet->GetActiveHDPubKey();
+            if (!extpub) {
+                throw JSONRPCError(RPC_WALLET_ERROR, "This wallet does not have an active xpub");
```

Comment:
in 9900207360a2852041ceed260c3430e92ff860c5

nit: "The wallet does not have an active **HD key**"

"active xpub" could mean xpub of active descriptor

Reply:
Done